docs/versioned_docs/version-zenith/developer-guide/python/944887-attribute-functions.mdx

```diff
@@ -0,0 +1,50 @@
+---
+slug: /developer-guide/python/944887/attribute-functions
+displayed_sidebar: "zenith"
+toc_max_heading_level: 2
+title: "Attributes as Functions"
+---
+
+import PartialExperimentalDocs from '../../partials/_experimental.mdx';
+
+# Attributes as Functions
+
+<PartialExperimentalDocs />
+
+Object attributes can be exposed as Dagger Functions, without having to
+create a getter function explicitly.
+
+There's a [dagger.field][dag-field] descriptor that is a wrapper of
+[dataclasses.field][data-field], but creates a getter function for the attribute
+as well so that it's accessible from the Dagger API.
```

Comment:
Thanks for this, @helderco . I believe you've already made the necessary changes in the Python docs, but you're suggesting that this could be replicated for the other language guides (please correct me if I'm wrong). In this case and as this is a technical detail that I'm not familiar with, I would prefer to have @jedevc @vito @TomChv to review this comment, also the comments in https://github.com/dagger/dagger/pull/6681#discussion_r1502631807 and https://github.com/dagger/dagger/pull/6681#discussion_r1502629268 and consider whether to make similar changes to the Go and TypeScript docs.

Reply:
I think it makes sense to explain the difference between object field and object function attributes in Go, but the dataclasses construction is python-specific.